agent/src/beerocks/slave/ap_manager_thread.cpp

```diff
@@ -708,6 +737,51 @@ bool ap_manager_thread::handle_cmdu(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_
         LOG(DEBUG) << "CLIENT_DISALLOW: mac = " << sta_mac << ", bssid = " << bssid;
 
         ap_wlan_hal->sta_deny(sta_mac, bssid);
+
+        // Check if validity period is set then add it to the "disallowed client timeouts" list
+        // This list will be polled in after_select()
+        // When validity period is timed out CLIENT_ALLOW message will be sent.
+        if (request->validity_period()) {
+
+            disallowed_client_params_t disallowed_client_params;
+            disallowed_client_params.mac   = request->mac();
+            disallowed_client_params.bssid = request->bssid();
+
+            // check if STA is already disallowed from this bssid
+            auto disallowed_client_timeout = std::find_if(m_disallowed_client_timeouts.begin(), m_disallowed_client_timeouts.end(),
+                        [&](const std::pair<std::chrono::steady_clock::time_point, disallowed_client_params_t> &element) {
+                            return ((element.second.mac == disallowed_client_params.mac) && (element.second.bssid == disallowed_client_params.bssid));
+                        });
+            if (disallowed_client_timeout != m_disallowed_client_timeouts.end()) {
+                // remove old disallow timeout from the list before inserting the new one
+                m_disallowed_client_timeouts.erase(disallowed_client_timeout);
+                LOG(DEBUG) << "remove old disallow timeout from the list before inserting the new one";
+            }
+
+            // calculate new disallow timeout from client validity period parameter [sec]
+            // when timeout expires allow message will be sent to the client.
+            auto disallow_timeout_in_sec =
+                std::chrono::steady_clock::now() + std::chrono::seconds(request->validity_period());
+            
+            LOG(DEBUG) << "insert new disallow timeout to the list";
+            // insert new disallow timeout to the list
+            m_disallowed_client_timeouts.insert(
+                std::pair<std::chrono::steady_clock::time_point, disallowed_client_params_t>(
+                    disallow_timeout_in_sec, disallowed_client_params));
```

Comment:
Wouldn't it be simpler to add the timeout to disallowed_client_params_t instead of using pairs?